breaker.go

```diff
@@ -0,0 +1,159 @@
+package resty
+
+import (
+	"errors"
+	"net/http"
+	"sync/atomic"
+	"time"
+)
+
+// CircuitBreaker can be in one of three states: Closed, Open, or Half-Open.
+// When the circuit breaker is Closed, requests are allowed to pass through.
+// If a failure count threshold is reached within a specified time-frame, the circuit breaker transitions to the Open state.
+// When the circuit breaker is Open, requests are blocked.
+// After a specified timeout, the circuit breaker transitions to the Half-Open state.
+// When the circuit breaker is Half-Open, a single request is allowed to pass through.
+// If that request fails, the circuit breaker returns to the Open state.
+// If SuccessThreshold requests succeed, the circuit breaker transitions back to the Closed state.
+type CircuitBreaker struct {
```

Comment:
@segevda, Please update the GoDoc with bullet points to explain specific details.

Reply:
I'm not sure which details are missing here, can you elaborate?

Reply:
I meant GoDoc formatting with bullet points for scenarios.

Reply:
Done